pkg/graveler/settings/manager.go

```diff
@@ -67,6 +68,21 @@ func (m *Manager) Save(ctx context.Context, repository *graveler.RepositoryRecor
 	return kv.SetMsg(ctx, m.store, graveler.RepoPartition(repository), []byte(graveler.SettingsPath(key)), setting)
 }
 
+// SaveIf persists the given setting under the given repository and key. Overrides settings key in KV Store.
+// The setting is persisted only if the given ETag matches the ETag of the current setting.
+// Otherwise, ErrPreconditionFailed is returned.
```

Comment:
Usually in our code when predicate (ifMatchETag) is nil - this is an indication for saveIf not exists. This might cause a bit of confusion

Reply:
Also, will this make the Save method redundant?

Reply:
Fixed (kind of): changed this to no longer be a pointer. The update is always conditional - empty string means update-if-not-exists. Now `branch.ProtectionManager` exposes both `SetRules` and `SetRulesIf`. SetRules uses the unconditional `Save`.

However, the catalog and graveler still use a pointer, where nil means unconditional update, and empty string means update-if-not-exists.